enhancements/cluster-logging/cluster-logging-log-forwarding.md

```diff
@@ -104,198 +156,160 @@ This is often required for industries such as the US public sector, healthcare o
 * A pipeline shall support at least one output.
 * A pipeline may support multiple outputs.
 * Outputs that require secrets which are not created by the cluster-logging-operator shall be created and managed by the administrator of the endpoint
-* Secrets shall have keys of: `tls.crt`, `tls.key`, `ca-bundler.crt` which point to the respective certificates for which they represent. Secrets shall have the key `shared_key` for use when using `forward` in a secure manner
-* The operator will only support a single instance of log forwarding named `instance` for at least the tech preview release
-* Log forwarding is enabled for tech preview when the instance of `clusterloggging` is annotated with: `clusterlogging.openshift.io/logforwardingtechpreview: enabled`
+* Forwarder CRs can refer to secrets for TLS certificates (see security section below)
+* The operator allows multiple instances of `forwarder`.
+  * Instances in the `openshift-logging` namespace are "global" and can forward logs from any container.
+  * Instances in other namespaces are namespace-restricted and can only forward logs from their own namespace.
+* Log forwarding is enabled when a `forwarder` object is created, and disabled when no forwarder objects exist.
+* When log forwarding is disabled, default forwarding to the internal elasticsearch instance is enabled.
+* When log forwarding is enabled, default forwarding is disabled. To enable both, create a global forwarder in the openshift-logging namespace with a single pipeline: `pipeline: { output: openshift-logging-default }`
 
 #### Security
-* A pipeline may support TLS configuration by declaring a secret. Secrets must contain keys: `tls.crt`, `tls.key`, `ca-bundler.crt`
-* The secret must exist in the `openshift-logging` namespace
-* Endpoints defined without a secret will use unsecure communication and must explicitly declare `insecure: true` in the output specifification
+* An output MAY support basic server-authenticated TLS by using a URL scheme that requires TLS, e.g. elasticsearch+https:// syslog+tls:// etc.
+* An output MAY support client-authenticated TLS by declaring a `secret` with keys `tls.crt`, `tls.key`, `ca.crt`
+* An output may support username/password or shared-key authentication either in the `url` or in the `secret`.
+* Outputs that intentionally do not use TLS must have an explicit `insecure: true` to avoid security breaches due to mis-configuration.
 
 #### Log Message Routing
-* Undeclared source types (e.g. `logs.infra`) will be dropped if they are not defined in `spec.pipelines` entry
-* Declared source types are routed to the specified output
-
-For example, If you have specified a pipeline for `logs.app` sources, but did not specify one for `logs.infra`, then `logs.infra` will be dropped
-
+* Pipelines can select multiple container log streams (fan in) by:
+  * Source type (`Infrastructure`, `Application`, `Audit`) 
+  * k8s label selector map or expressions.
```

Comment:
What expressions are meant here? Does this mean `FieldSelectors`, too? I would be explicit here. IIRC label selectors allow only `=`, `==` and `!=`.

Reply:
Gaahh, see above. Need to revisit the selector design when we're ready for it. Label selectors come in two kinds: maps (called "equality-based") and expression strings (called "set-based")  Set based selector expressions have other operators 'in', 'notin' etc. Frankly the k8s design here is a surprising mess, given that George Boole explained how to do it over 200 years ago, but we have to find a way to make it work and be consistent with general practice.

Reply:
Also re: "AFAIU, this boils down to add arbitrary non-transport-layer configuration per output to the collector configuration. I have the impression that there is a bias towards fluentd configuration. If we would add this extra bit, would this work with other collectors, e.g. promtail, too?"
In the current API, the "extra bit" is a union and we will add options as we go. The goal is to be able to configure most forwarders *without* using this union, but it provides an "escape" for essential settings that are type-specific. For example a syslog output might need to set the 'facility', and there's no generic way to do that so the syslog map allows it. Anything you set in one of these type-specific maps will not work for other output types, that's why we want to make the majority use cases work with generic